### PR TITLE
Introduce a method to get OIDC scopes registered in the tenant.

### PR DESCRIPTION
**Pull request by shashimalcse**
### Proposed changes in this pull request
$subject

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 1]
```diff
@@ -1863,4 +1865,32 @@ private void validateScopeExistence(String scopeName) throws IdentityOAuthAdminE
...
+     */
+    public List<String> getRegisteredOIDCScope(String tenantName) throws IdentityOAuthAdminException {
+
+
+        try {
+            int tenantId = getTenantId(tenantName);
+            List<String> oidcScopes = OAuthTokenPersistenceFactory.getInstance().getScopeClaimMappingDAO().
+                    getScopeNames(tenantId);
+
+            if (CollectionUtils.isEmpty(oidcScopes)) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:1884]: Is there a possibility for DAO to return null?
if not we can redirectly return the result from DAO

**Review by mefarazath — DISMISSED** [review]

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 2]
```diff
@@ -1863,4 +1864,21 @@ private void validateScopeExistence(String scopeName) throws IdentityOAuthAdminE
...
+     * @return List of OIDC scopes registered in tenant.
+     * @throws IdentityOAuthAdminException exception if OIDC scope retrieval fails.
+     */
+    public List<String> getRegisteredOIDCScope(String tenantDomain) throws IdentityOAuthAdminException {
+
+        try {
+            int tenantId = getTenantId(tenantDomain);
+            return  OAuthTokenPersistenceFactory.getInstance().getScopeClaimMappingDAO().getScopeNames(tenantId);
+        } catch (IdentityOAuth2Exception e) {
+            throw handleError("Error while loading OIDC scopes and claims for tenant: " + tenantDomain, e);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:1881]:
```suggestion
            throw handleError("Error while loading OIDC scope names for tenant: " + tenantDomain, e);
```

**Review by mefarazath — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 3]
```diff
@@ -1863,4 +1864,21 @@ private void validateScopeExistence(String scopeName) throws IdentityOAuthAdminE
                             scopeName));
         }
     }
+
+    /**
+     * Returns OIDC scopes registered in the tenant.
+     *
+     * @param tenantDomain tenant domain
```
Comment (sandushi) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:1871]: Shall we put a full stop at the end of this comment?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 4]
```diff
@@ -4621,34 +4620,4 @@ public static boolean isFederatedRoleBasedAuthzEnabled(String clientId) throws I
         }
         return isFederatedRoleBasedAuthzEnabled;
     }
-
-    public static String[] getRequestedOIDCScopes(String[] requestedScopes)
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:4625]: why are we removing a public method? Ideally, we should deprecate in the first release and then gradually we should remove it?

**Pull request merged**